src/accelerate/data_loader.py

```diff
@@ -606,7 +606,16 @@ def _fetch_batches(self, iterator):
                     batches = []
                     for _ in range(self.state.num_processes):
                         batches.append(next(iterator))
-                    batch = concatenate(batches, dim=0)
+                    try:
+                        batch = concatenate(batches, dim=0)
+                    except RuntimeError:
+                        raise RuntimeError(
```

Comment:
```suggestion
                    except RuntimeError as e:
                        raise RuntimeError(
```

and have `from e` at the end.

Reply:
This makes the quality check red. You you can to print the error message somewhere ?